browser/ipfs/content_browser_client_helper_unittest.cc

```diff
@@ -137,4 +138,39 @@ TEST_F(ContentBrowserClientHelperUnitTest, HandleIPNSURLRewriteLocal) {
   ASSERT_TRUE(HandleIPFSURLRewrite(&ipns_uri, browser_context()));
 }
 
+TEST_F(ContentBrowserClientHelperUnitTest, HandleIPFSURLReverseRewriteLocal) {
+  profile()->GetPrefs()->SetInteger(
+      kIPFSResolveMethod, static_cast<int>(IPFSResolveMethodTypes::IPFS_LOCAL));
+  ASSERT_EQ(ipfs::GetConfiguredBaseGateway(browser_context(),
+    version_info::Channel::UNKNOWN),
+      GURL("http://localhost:48080/"));
+  GURL ipns_uri("http://test.com.ipns.localhost:8000");
+  ASSERT_FALSE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
+  ipns_uri = GURL("http://test.com.ipns.localhost:48080");
+  ASSERT_TRUE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
+}
+
+TEST_F(ContentBrowserClientHelperUnitTest, HandleIPFSURLReverseRewriteGateway) {
+  profile()->GetPrefs()->SetInteger(kIPFSResolveMethod,
+    static_cast<int>(IPFSResolveMethodTypes::IPFS_GATEWAY));
+  ASSERT_EQ(ipfs::GetConfiguredBaseGateway(browser_context(),
+    version_info::Channel::UNKNOWN), GURL("https://dweb.link/"));
+  GURL ipns_uri("http://test.com.ipns.localhost:8000");
+  ASSERT_FALSE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
+  ipns_uri = GURL("https://ku2jvrakgpiqgx4j6fe.ipfs.dweb.link");
+  ASSERT_TRUE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
+
+  profile()->GetPrefs()->SetString(kIPFSPublicGatewayAddress,
+                                   "http://localhost:8080");
+  ASSERT_EQ(ipfs::GetConfiguredBaseGateway(browser_context(),
+                                           version_info::Channel::UNKNOWN),
+            GURL("http://localhost:8080"));
+  ipns_uri = GURL("http://test.com.ipns.localhost:8000");
+  ASSERT_FALSE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
+  ipns_uri = GURL("https://ku2jvrakgpiqgx4j6fe.ipfs.dweb.link");
+  ASSERT_FALSE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
+  ipns_uri = GURL("http://test.com.ipns.localhost:8080");
+  ASSERT_TRUE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
```

Comment:
Please also check the new value of `ipns_uri` in each of these tests and the ones in `HandleIPFSURLReverseRewriteLocal` too.